From 9d2605e51b9ba83382c5da3a838656c9910d75a1 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Tue, 11 Nov 2014 21:18:12 +0800
Subject: [PATCH] MLK-9833-2 dts: imx6sx: do not touch CAN gpios pads if M4 is
 enabled

CAN devices are allocated to run on M4.
So do not touch CAN pads setting if M4 is enabled.

Conflicts:

	arch/arm/boot/dts/imx6sx-sdb.dts

Signed-off-by: Dong Aisheng <b29396@freescale.com>
(cherry picked from commit 4fc10d386d02e8ae10d8512fdadb5a02ddd25228)
---
 arch/arm/boot/dts/imx6sx-sdb-m4.dts | 5 +++++
 arch/arm/boot/dts/imx6sx-sdb.dts    | 9 ++++++++-
 2 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx6sx-sdb-m4.dts b/arch/arm/boot/dts/imx6sx-sdb-m4.dts
index 5309be9..d2c6a67 100644
--- a/arch/arm/boot/dts/imx6sx-sdb-m4.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb-m4.dts
@@ -15,6 +15,11 @@
 	};
 };
 
+&iomuxc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog>;
+};
+
 /*
  * The flollowing modules are conflicting with M4, disable them when m4
  * is running.
diff --git a/arch/arm/boot/dts/imx6sx-sdb.dts b/arch/arm/boot/dts/imx6sx-sdb.dts
index 8a5a9ef..ecfb7af 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb.dts
@@ -683,7 +683,7 @@
 
 &iomuxc {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hog>;
+	pinctrl-0 = <&pinctrl_hog &pinctrl_can_gpios>;
 
 	imx6x-sdb {
 		pinctrl_hog: hoggrp {
@@ -806,6 +806,13 @@
 			>;
 		};
 
+		pinctrl_can_gpios: can-gpios {
+			fsl,pins = <
+				MX6SX_PAD_QSPI1B_DATA1__GPIO4_IO_25 0x17059
+				MX6SX_PAD_QSPI1B_DATA3__GPIO4_IO_27 0x17059
+			>;
+		};
+
 		pinctrl_i2c1: i2c1grp {
 			fsl,pins = <
 				MX6SX_PAD_GPIO1_IO01__I2C1_SDA 0x4001b8b1
-- 
1.8.0

